typed-racket-lib/typed-racket/typecheck/tc-lambda-unit.rkt

```diff
@@ -43,14 +43,18 @@
 ;; (i.e. (append positional (or id '())) but syntax)
 (struct formals (positional rest syntax) #:transparent)
 
-(define (make-formals stx)
+(define (make-formals stx [not-in-poly #t])
```

Comment:
This needs a comment explaining what's going on with `not-in-poly`

Reply:
I think what's going on here is that we want to remove some information that might have out-of-scope type variables in particular cases. But the comment should explain why.